.github/workflows/auto_update_charm_libs.yaml

```diff
@@ -0,0 +1,42 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+name: Auto-update Charm Libraries
+on:
+  workflow_call:
+
+permissions:
```

Comment:
Do you think it should have something about it in the README since it requires contents/pull-requests permissions?